service_configuration_lib/spark_config.py

```diff
@@ -93,19 +106,27 @@ def get_mesos_spark_env(
         spark_env['spark.mesos.uris'] = 'file:///root/.dockercfg'
 
     spark_env = {**spark_env, **user_spark_opts}
-    spark_env['spark.mesos.executor.docker.parameters'] = 'cpus={}'.format(spark_env['spark.executor.cores'])
+    spark_env['spark.mesos.executor.docker.parameters'] = 'cpus={},{}'.format(
+        spark_env['spark.executor.cores'],
+        spark_env['spark.mesos.executor.docker.parameters'],
+    )
     _validate_spark_env(spark_env, event_log_dir)
 
     return spark_env
 
 
+def _generate_volume_name(volume):
```

Comment:
I'm unconvinced that there's value in having this as a function, can't we just call `volume['containerPath']` directly where we need it?

Reply:
In case we need to change the format of volume name to use. Current the option will look like `spark.kubernetes.executor.volumes.hostPath./nail/etc/beep.mount.path`, nothing wrong with it but later we might want to modify to increase readability, like `/nail/etc` to `nail-etc`.